universal-application-tool-0.0.1/app/views/applicant/ApplicantProgramSummaryView.java

```diff
@@ -153,36 +163,55 @@ private ContainerTag renderQuestionSummary(AnswerData data, Long applicantId) {
       answerContent.withText(texts[i]);
     }
 
-    // Link to block containing specific question.
-    String editLink =
-        routes.ApplicantProgramBlocksController.review(
-                applicantId, data.programId(), data.blockId())
-            .url();
-    ContainerTag editAction =
-        new LinkElement()
-            .setHref(editLink)
-            .setText("Edit")
-            .setStyles(Styles.ABSOLUTE, Styles.BOTTOM_0, Styles.RIGHT_0)
-            .asAnchorText();
-    ContainerTag editContent =
-        div(editAction)
-            .withClasses(
-                Styles.FLEX_AUTO,
-                Styles.TEXT_RIGHT,
-                Styles.FONT_LIGHT,
-                Styles.ITALIC,
-                Styles.RELATIVE);
     ContainerTag answerDiv =
-        div(answerContent, editContent).withClasses(Styles.FLEX, Styles.FLEX_ROW, Styles.PR_2);
+        div(answerContent).withClasses(Styles.FLEX, Styles.FLEX_ROW, Styles.PR_2);
+
+    // Maybe link to block containing specific question.
+    if (inReview || isAnswered || isFirstUnanswered) {
+      String editText =
+          (!isAnswered && !inReview)
+              ? messages.at(MessageKey.LINK_BEGIN.getKeyName())
+              : messages.at(MessageKey.LINK_EDIT.getKeyName());
+
+      String editLink =
+          (!isAnswered && !inReview)
+              ? routes.ApplicantProgramBlocksController.edit(
+                      applicantId, data.programId(), data.blockId())
+                  .url()
+              : routes.ApplicantProgramBlocksController.review(
+                      applicantId, data.programId(), data.blockId())
+                  .url();
+
+      ContainerTag editAction =
+          new LinkElement()
+              .setHref(editLink)
+              .setText(editText)
+              .setStyles(
+                  Styles.ABSOLUTE,
```

Comment:
I'm pretty hesitant about absolute style 😛  What are your thoughts on using flexbox or grid to put this Edit button here? Non-blocking, because it looks like it works and we don't need to change this just for the heck of it! But I often find flexbox and grid very delightful to work with.

Reply:
Ohh, maybe I misunderstood how absolute is being used here because I see flex used above and below. Disregard if so.